### PR TITLE
- Add list-events support in alerts-api

### PR DESCRIPTION
**Pull request by jshaughn**
- Add to the spec
  - note, see mechanism for ignoring parts of the uri in the vcr match
- refactor generate_query_params
  - not really necessary, mainly a ruby exercise for me

### CODE REVIEW

File: lib/alerts/alerts_api.rb [hunk 1]
```diff
@@ -67,9 +69,11 @@ def get_alerts_for_trigger(trigger_id) # TODO: add additional filters
     end
 
     # List fired alerts
+    # @param [Hash]criteria optional query criteria
```
Comment (pilhuhn) [inline lib/alerts/alerts_api.rb:72]: Can you add the key, value types for the hash please?

File: lib/alerts/alerts_api.rb [hunk 2]
```diff
@@ -109,6 +113,17 @@ def acknowledge_alert(alert_id, by = nil, comment = nil)
...
       true
     end
+
+    # List Events given optional criteria
+    # @param [Hash] criteria optional query criteria
+    # @return [Array<Event>] List of all alerts in the system. Can be empty
+    def list_events(*criteria)
+      query = generate_query_params(*criteria)
+      ret = http_get('/events' + query)
+      val = []
```
Comment (pilhuhn) [inline lib/alerts/alerts_api.rb]: Try to use `ret.map{..}`. I know I've used this current way also (too often).

**Comment by pilhuhn** [comment]
Two small comments - otherwise looks good to me :+1:

**Comment by jshaughn** [comment]
I amended the prior commit and force-pushed, so comment lines may be off, but the changes are in.

**Pull request merged**